src/tests/utests/iotc_utest_bsp_crypto_ecc.cc

```diff
@@ -0,0 +1,259 @@
+  uint8_t ecc_signature_new[64];
+  size_t bytes_written = 0;
+
+  EXPECT_EQ(iotc_bsp_ecc(&DefaultPrivateKey, ecc_signature,
+                         kEccSignatureBufferLength, &bytes_written,
+                         kDefaultDataToSign, kDefaultDataToSignLength),
+            IOTC_BSP_CRYPTO_BUFFER_TOO_SMALL_ERROR);
+  // Two 32-byte integers(r and s) build up a JWT ECC signature
+  EXPECT_EQ(bytes_written, 64u);
+
+  const size_t kEccSignatureNewBufferLen = bytes_written;
+
+  // by definition, the expected JWT ECC signature size is 64 bytes
+  // https://tools.ietf.org/html/rfc7518#section-3.4
+  ASSERT_EQ(64u, kEccSignatureNewBufferLen);
+  EXPECT_EQ(iotc_bsp_ecc(&DefaultPrivateKey, ecc_signature_new,
+                         kEccSignatureNewBufferLen, &bytes_written,
+                         kDefaultDataToSign, kDefaultDataToSignLength),
+            IOTC_BSP_CRYPTO_STATE_OK);
+  EXPECT_EQ(bytes_written, kEccSignatureNewBufferLen);
+}
+
+TEST_F(IotcBspCryptoEcc, ReportsErrorOnInvalidPrivateKey) {
+  constexpr char kInvalid[] = "invalid key";
+  const iotc_crypto_private_key_data_t kInvalidKey = {
+      IOTC_CRYPTO_KEY_UNION_TYPE_PEM, const_cast<char *>(kInvalid),
+      IOTC_JWT_PRIVATE_KEY_SIGNATURE_ALGORITHM_ES256};
+  const size_t kEccSignatureBufferLength = 128;
+  uint8_t ecc_signature[kEccSignatureBufferLength];
+  size_t bytes_written = 0;
+
+  EXPECT_EQ(iotc_bsp_ecc(&kInvalidKey, ecc_signature, kEccSignatureBufferLength,
+                         &bytes_written, kDefaultDataToSign,
+                         kDefaultDataToSignLength),
+            IOTC_BSP_CRYPTO_KEY_PARSE_ERROR);
+}
+
+TEST_F(IotcBspCryptoEcc, ReportsErrorWhenOutputBufferIsNull) {
+  size_t bytes_written = 0;
+
+  EXPECT_EQ(iotc_bsp_ecc(&DefaultPrivateKey, nullptr,
+                         /*ecc_signature_buffer_len=*/128, &bytes_written,
+                         kDefaultDataToSign, kDefaultDataToSignLength),
+            IOTC_BSP_CRYPTO_INVALID_INPUT_PARAMETER_ERROR);
+}
+
+TEST_F(IotcBspCryptoEcc, ReportsErrorWhenInputBufferIsNull) {
+  const size_t kEccSignatureBufferLength = 128;
+  uint8_t ecc_signature[kEccSignatureBufferLength];
+  size_t bytes_written = 0;
+
+  EXPECT_EQ(iotc_bsp_ecc(&DefaultPrivateKey, ecc_signature,
+                         kEccSignatureBufferLength, &bytes_written, nullptr,
+                         kDefaultDataToSignLength),
+            IOTC_BSP_CRYPTO_INVALID_INPUT_PARAMETER_ERROR);
+}
+
+TEST_F(IotcBspCryptoEcc, JwtSignatureValidation) {
+  const char *kJwtHeaderPayloadB64 =
+      R"(eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.
+      eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWUs
+      ImlhdCI6MTUxNjIzOTAyMn0)";
+  uint8_t hash_sha256[32] = {0};
+  iotc_bsp_sha256(hash_sha256, (const uint8_t *)kJwtHeaderPayloadB64,
+                  strlen(kJwtHeaderPayloadB64));
+  size_t bytes_written_ecc_signature = 0;
+  uint8_t ecc_signature[IOTC_JWT_MAX_SIGNATURE_SIZE] = {0};
+
+  size_t i = 0;
```

Comment:
This can go into the for loop, no? You don't use it outside.

for (size_t i =0; i < IOTC_JWT_MAX_SIGNATURE_SIZE; ++i) {
 ..
}